### PR TITLE
Load performance test pipelines from files

### PR DESCRIPTION
**Pull request by ckadner**
**Description of your changes:**

Updates to the performance test script:

* Move the the `performance_test.py` outside of the `compiler` folder (since the scope is beyond compiler testing) 
* Add `perf_test_config.yaml` with paths of test pipelines
* Add a `Status` column to output CSV file
* Update `compile_pyfile` to remove loaded pipeline module after compilation to allow subsequent imports of another module with the same name

**Output:**

| Pipeline | Status | Compile\_Time | Submit\_Time | Run\_Time | Num\_TaskRuns | Num\_Runs | Total\_TaskRun\_Time | Total\_Run\_Time |
| :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- |
| trusted-ai | Failed | 0:00:00.370032 | 0:00:08.471185 | 0:05:27.214805 | 1 | 0 | 0:04:26 | 0:00:00 |
| calc\_pipeline | Succeeded | 0:00:00.163394 | 0:00:08.852992 | 0:11:05.103077 | 3 | 0 | 0:08:07 | 0:00:00 |
| Flip-Coin-Custom-Task | Completed | 0:00:00.114080 | 0:00:09.203534 | 0:11:27.995799 | 3 | 2 | 0:08:11 | 0:00:00 |
| Flip-Coin-samples | Completed | 0:00:00.119328 | 0:00:08.908992 | 0:14:10.104801 | 7 | 0 | 0:12:36 | 0:00:00 |
| withitem\_nested | Succeeded | 0:00:00.049208 | 0:00:08.116537 | 0:17:57.169451 | 1 | 1 | 0:04:20 | 0:17:27 |

**Which issue is resolved by this Pull Request:** 

Resolves: #778

/cc @Tomcli
/cc @yhwang

### CODE REVIEW

**Comment by Tomcli** [comment]
thanks @ckadner, I also made few changes to the script mainly to update the examples to run for performance tests and fix some bugs. #795

File: sdk/python/tests/perf_test_config.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#  - name: "Loop-Static"
+#    path: "sdk/python/tests/compiler/testdata/loop_static.py"
+#  - name: "Conditions-and-Loops"
+#    path: "sdk/python/tests/compiler/testdata/conditions_and_loops.py"
+#  - name: "With-item-Nested"
+#    path: "sdk/python/tests/compiler/testdata/withitem_nested.py"
+#  - name: "Condition-and-recur"
+#    path: "sdk/python/tests/compiler/testdata/cond_recur.py"
+  - name: "Flip-Coin-samples"
+    path: "samples/flip-coin/condition.py"               # condition with simple python
```
Comment (Tomcli) [inline sdk/python/tests/perf_test_config.yaml:33]: We may want to update this list of example. I figured out is better to run performance tests on sequential pipelines so we can calculate the hidden cost for scheduling and storing tasks.

Comment (Tomcli) [reply sdk/python/tests/perf_test_config.yaml:33]: It is part of my #795 PR

File: sdk/python/kfp_tekton/compiler/main.py [hunk 2]
```diff
@@ -82,6 +82,7 @@ def compile_pyfile(pyfile, function_name, output_path, type_check, tekton_pipeli
     _compile_pipeline_function(pipeline_funcs, function_name, output_path, type_check, tekton_pipeline_conf)
   finally:
     del sys.path[0]
+    sys.modules.pop(os.path.splitext(filename)[0])
```
Comment (ckadner) [inline sdk/python/kfp_tekton/compiler/main.py:85]: Since Python caches previsously loaded modules, this change was required to allow using the `compile_pyfile()` method multiple times within the same Python process and enable loading different pipeline scripts from files with the same ("module") name, i.e. we have multiple `condition.py` samples.

Without this change, we would get these errors:
```
ValueError trying to compile /Users/ckadner/PycharmProjects/kfp-tekton_ckadner/samples/flip-coin/condition.py: A function with @dsl.pipeline decorator is required in the py file.
ValueError trying to compile /Users/ckadner/PycharmProjects/kfp-tekton_ckadner/samples/flip-coin-custom-task/condition.py: A function with @dsl.pipeline decorator is required in the py file.
```

File: sdk/python/tests/perf_test_config.yaml (hunk 1 above)
Comment (ckadner) [reply sdk/python/tests/perf_test_config.yaml:33]: quite a delicate dance of well-timed merges we have ahead of us here ;-)

File: sdk/python/tests/performance_tests.py [hunk 3]
```diff
@@ -205,91 +224,111 @@ def get_client() -> TektonClient:
...
+
+    assert script_path_actually == project_root_folder.joinpath(script_path_presumed), \
+        "Can not determine project root folder. Was this script file moved or renamed?"
+
+    return project_root_folder
+
+
+def load_test_config() -> dict:
 
-    from testdata.sequential import sequential_pipeline
```
Comment (ckadner) [inline sdk/python/tests/performance_tests.py:211]: the diff shown here is a little misleading, GH tries to align lines of code that are not related

File: sdk/python/tests/performance_tests.py [hunk 4]
```diff
@@ -205,91 +224,111 @@ def get_client() -> TektonClient:
     return client
 
 
-def load_pipeline_functions() -> [(Callable, str)]:
```
Comment (ckadner) [inline sdk/python/tests/performance_tests.py:208]: the diff shown from here on down is a little misleading, GH tries to align lines of code that are not related

File: sdk/python/tests/performance_tests.py [hunk 5]
```diff
@@ -205,91 +224,111 @@ def get_client() -> TektonClient:
...
+        pipeline_files_with_name.append((fp, name))
 
-def run_concurrently(pipelinefunc_name_tuples: [(Callable, str)]) -> [(str, str)]:
-    pipeline_status = []
+    print(f"Loaded {len(pipeline_files_with_name)} pipelines from {TEST_CONFIG}\n")
+
+    return pipeline_files_with_name
+
+
+def run_concurrently(pipelinescript_name_tuples: [(Path, str)]):
```
Comment (ckadner) [inline sdk/python/tests/performance_tests.py:278]: from here on down the GH diff alignment works again

**Comment by ckadner** [comment]
@Tomcli -- sorry for the concurrent PRs and difficult merging process. I hope by adding the changes you made to `performance_tests.py` here in my PR, we can merge this PR first and then merge your PR without the `performance_tests.py` changes to avoid messy merge conflict resolutions

**Comment by ckadner** [comment]
Thanks @Tomcli
